packages/repository/src/repositories/relation.factory.ts

```diff
@@ -4,21 +4,23 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {EntityCrudRepository} from './repository';
-import {RelationType} from '../decorators/relation.decorator';
+import {
+  RelationType,
+  HasManyDefinition,
+  RelationDefinitionBase,
+} from '../decorators/relation.decorator';
 import {Entity} from '../model';
 import {
-  HasManyEntityCrudRepository,
+  HasManyRepository,
   DefaultHasManyEntityCrudRepository,
 } from './relation.repository';
+import {Class} from '../common-types';
+import {RelationMap} from '../decorators/model.decorator';
 
-export interface RelationDefinitionBase {
-  type: RelationType;
-  keyTo: string;
-}
+export type HasManyRepositoryFactory<S extends Entity, T extends Entity> = (
+  key: Partial<S>,
```

Comment:
Assuming pk is `id` and fk is `categoryId`, I thought the constraint here would look like `{id: someValue}`. Is it supposed to look like `{categoryId: someValue}`?

Reply:
How confusing!

Ideally, I'd like to pass the pk/fk value only and don't have to worry whether the constraint is for the source or the target model.

Personally, I find it more natural to express the constraint using target property names because it's closer to how the actual constraint will be implemented. For example, I can imagine a constraint like `{authorId: someValue, ownerId: someOtherValue}`. I find it difficult to imagine how could we implement a constraint on the source model considering that the query is going to be made against the target model and we don't support JOIN statements that would allow us to construct a query along the lines of `SELECT * FROM target LEFT JOIN source ON target.categoryId = source.id WHERE source.someProperty = someValue`.